### PR TITLE
fix compatibility with the new dependency

### PR DESCRIPTION
**Pull request by geoffreyp**
fix compatibility : 
 - connect before each call because the light seems to be in "sleep mode"
 - force to disconnect after call because we can't be connected in more than twice sessions

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
       "resolved": "https://registry.npmjs.org/async/-/async-2.6.1.tgz",
       "integrity": "sha512-fNEiL2+AZt6AlAw/29Cr0UDe4sRAHCpEHh54WMz+Bb7QfNcFw4h3loofyJpLeQs4Yx7yuqu/2dLgM5hKOs6HlQ==",
       "requires": {
-        "lodash": "4.17.11"
+        "lodash": "^4.17.10"
```
Comment (NicolasD-62) [inline package-lock.json]: Why use the previous version?

File: package-lock.json [hunk 2]
```diff
@@ -45,18 +45,18 @@
       "resolved": "https://registry.npmjs.org/portscanner/-/portscanner-2.2.0.tgz",
       "integrity": "sha512-IFroCz/59Lqa2uBvzK3bKDbDDIEaAY8XJ1jFxcLWTqosrsc32//P4VuSB2vZXoHiHqOmx8B5L5hnKOxL/7FlPw==",
       "requires": {
-        "async": "2.6.1",
-        "is-number-like": "1.0.8"
+        "async": "^2.6.0",
```
Comment (NicolasD-62) [inline package-lock.json]: Why use the previous version?

File: package-lock.json [hunk 3]
```diff
@@ -45,18 +45,18 @@
...
+        "is-number-like": "^1.0.3"
```
Comment (NicolasD-62) [inline package-lock.json]: Why use the previous version?

File: lib/exec.js [hunk 4]
```diff
@@ -1,44 +1,38 @@
...
-	if (yeelight) {
+	yee.connect().then((yeelight) => {
 		switch (params.deviceType.type) {
 			case 'binary':
 				return yeelight.setPower(value == 1, 'smooth')
 					.catch((err) => Promise.reject(err));
 
 			case 'brightness':
-				return yeelight.setBright(value, 'smooth')
+				return yeelight.setBright(value, 'sudden')
```
Comment (NicolasD-62) [inline lib/exec.js]:
```javascript
yeelight.setBright(value, 'smooth', 500)
``` 
will be smoother and more stylish

File: lib/exec.js [hunk 5]
```diff
@@ -1,44 +1,38 @@
...
-						.then((currentSat) => {
-							return yeelight.setHSV(value, currentSat, 'smooth')
-								.catch((err) => Promise.reject(err));
-						});
-				} else {
-					return Promise.reject();
-				}
+				return utils.getLastState(params.deviceType.device, 'saturation')
+					.then((currentSat) => {
+						return yeelight.setHSV(value, currentSat, 'sudden')
```
Comment (NicolasD-62) [inline lib/exec.js]:
```javascript
yeelight.setHSV(value, currentSat, 'smooth', 500)
``` 
will be smoother and more stylish

File: lib/exec.js [hunk 6]
```diff
@@ -1,44 +1,38 @@
...
-						.then((currentHue) => {
-							return yeelight.setHSV(currentHue, value, 'smooth')
-								.catch((err) => Promise.reject(err));
-						});
-				} else {
-					return Promise.reject();
-				}
+				return utils.getLastState(params.deviceType.device, 'hue')
+					.then((currentHue) => {
+						return yeelight.setHSV(currentHue, value, 'sudden')
```
Comment (NicolasD-62) [inline lib/exec.js]:
```javascript
yeelight.setHSV(currentHue, value, 'smooth', 500)
``` 
will be smoother and more stylish

File: lib/exec.js [hunk 7]
```diff
@@ -1,44 +1,38 @@
...
+							.catch((err) => Promise.reject(err));
+					});
 		}
-	} else {
-		return Promise.reject();
-	}
+	})
+		.then((res) => yee.disconnect())
```
Comment (NicolasD-62) [inline lib/exec.js]: If switch is bypassed, there is no return, so maybe no disconnect...

File: lib/exec.js (hunk 4 above)
Comment (geoffreyp) [reply lib/exec.js]: smooth on "brightness" create a big latence on my lamp

File: lib/exec.js (hunk 6 above)
Comment (geoffreyp) [reply lib/exec.js]: like for "brightness", it create a big latence on my lamp

File: package-lock.json (hunk 1 above)
Comment (geoffreyp) [reply package-lock.json]: problem with my git IDE, I fix it, sorry

File: lib/exec.js (hunk 7 above)
Comment (geoffreyp) [reply lib/exec.js]: Exact ! I add a default case to not bypass disconnect()

File: lib/exec.js (hunk 4 above)
Comment (geoffreyp) [reply lib/exec.js]: I added the duration and it works !

File: lib/exec.js (hunk 6 above)
Comment (geoffreyp) [reply lib/exec.js]: I added the duration and it works !

File: lib/exec.js (hunk 5 above)
Comment (geoffreyp) [reply lib/exec.js]: I added the duration and it works !

**Pull request merged**